rpc/cloudagent/compute/placementgroup/moc_cloudagent_placementgroup.proto

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+
+syntax = "proto3";
+option go_package = "github.com/microsoft/moc/rpc/cloudagent/compute";
+package moc.cloudagent.compute;
+
+import "google/protobuf/wrappers.proto";
+import "moc_common_common.proto";
+
+enum PlacementGroupType {
+	Affinity = 0;
+	AntiAffinity = 1 ;
+	StrictAntiAffinity = 2;
+}
+
+enum PlacementGroupScope {
+	Server = 0;
+	Zone = 1 ;
+}
+
+message PlacementGroupRequest {
+	repeated PlacementGroup PlacementGroups = 1;
+	Operation OperationType = 2;
+}
+
+message PlacementGroupResponse {
+	repeated PlacementGroup PlacementGroups = 1;
+	google.protobuf.BoolValue Result = 2;
+	string  Error = 3;
+}
+
+message PlacementGroupPrecheckRequest {
+	repeated PlacementGroup PlacementGroups = 1;
+}
+
+message PlacementGroupPrecheckResponse {
+	// The precheck result: true if the precheck criteria is passed; otherwise, false
+	google.protobuf.BoolValue Result = 1;
+
+	// The error message if the precheck is not passed; otherwise, empty string
+	string Error = 2;
+}
+
+message VirtualMachineRef {
+	string groupName = 1;
```

Comment:
Nit: Is group name required for VM ref  ? since VMs can be added to PG in same group

Reply:
updated the placementgroup proto to resuse the VirtualMachineReference from availabilityset proto. so this comment is not more valid with new changes